### PR TITLE
CB-5712 adds loaders for administration menu items

### PR DESCRIPTION
**Pull request by sergeyteleshev**

### CODE REVIEW

File: webapp/packages/plugin-administration/src/Administration/getLoaders.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    return acc;
+  }
+
+  const loader = item.getLoader();
+
+  if (Array.isArray(loader)) {
+    acc.push(...loader);
+  } else {
+    acc.push(loader);
+  }
```
Comment (Wroud) [inline webapp/packages/plugin-administration/src/Administration/getLoaders.ts]: can be simpler: `return [...acc, item.getLoader?.() || []].flat()` (at this point do we need special function for that?)

File: webapp/packages/core-authentication/src/AuthProvidersResource.ts [hunk 2]
```diff
@@ -92,6 +92,8 @@ export class AuthProvidersResource extends CachedMapResource<string, AuthProvide
   }
 
   protected async loader(originalKey: ResourceKey<string>): Promise<Map<string, AuthProvider>> {
+    await this.serverConfigResource.load();
+
```
Comment (Wroud) [inline webapp/packages/core-authentication/src/AuthProvidersResource.ts]: it's better to use `this.preload(this.serverConfigResource, () => undefined)`

File: webapp/packages/core-authentication/src/AuthProvidersResource.ts [hunk 3]
```diff
@@ -92,7 +92,7 @@ export class AuthProvidersResource extends CachedMapResource<string, AuthProvide
   }
 
   protected async loader(originalKey: ResourceKey<string>): Promise<Map<string, AuthProvider>> {
-    await this.serverConfigResource.load();
+    this.preloadResource(this.serverConfigResource, () => undefined);
```
Comment (Wroud) [inline webapp/packages/core-authentication/src/AuthProvidersResource.ts]: proloadResource must be executed in constructor and only once

**Review by Wroud — CHANGES_REQUESTED** [review]

File: webapp/packages/core-authentication/src/AuthProvidersResource.ts [hunk 4]
```diff
@@ -44,6 +44,7 @@ export class AuthProvidersResource extends CachedMapResource<string, AuthProvide
   ) {
     super();
 
+    this.preloadResource(this.serverConfigResource, () => undefined);
     this.sync(
       serverConfigResource,
```
Comment (Wroud) [inline webapp/packages/core-authentication/src/AuthProvidersResource.ts]: sync already preloads resource

Comment (Wroud) [reply webapp/packages/core-authentication/src/AuthProvidersResource.ts]: so you don't need to add `this.preloadResource(this.serverConfigResource, () => undefined);` or load `serverConfigResource` in loader manually

**Review by Wroud — CHANGES_REQUESTED** [review]

**Review by Wroud — APPROVED** [review]

**Review by devnaumov — APPROVED** [review]

**Pull request merged**